src/main/java/seedu/address/logic/commands/RemarkCommand.java

```diff
@@ -0,0 +1,99 @@
+package seedu.address.logic.commands;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_REMARK;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_COMPANIES;
+
+import java.util.List;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.company.Company;
+import seedu.address.model.company.Remark;
+
+/**
+ * Adds a remark to an existing company in the address book.
+ */
+public class RemarkCommand extends Command {
+
+    public static final String COMMAND_WORD = "remark";
+    public static final String REMOVE_COMMAND_WORD = "unremark";
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the remark of the company identified "
+            + "by the index number used in the displayed company list. "
+            + "Existing remark will be overwritten by the input.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + PREFIX_REMARK + "[REMARK]\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_REMARK + "Likes to swim.";
+    public static final String MESSAGE_REMOVE_USAGE = REMOVE_COMMAND_WORD
+            + ": Removes the remark identified by the index number used in the displayed company list.\n"
+            + "Parameters: INDEX (must be a positive integer)\n"
+            + "Example: " + COMMAND_WORD + " 1";
+
+    public static final String MESSAGE_NO_REMARK = "Remarks missing in entry.";
+    public static final String MESSAGE_ADD_REMARK_SUCCESS = "Added remark to Company: %1$s";
+    public static final String MESSAGE_DELETE_REMARK_SUCCESS = "Removed remark from Company: %1$s";
+
+    private final Index index;
+    private final Remark remark;
+
+    /**
+     * @param index of the company in the filtered company list to edit the remark
+     * @param remark of the company to be updated to
+     */
```

Comment:
Missing description of constructor. 

Should have fullstops after parameters description.